connector/signaltometricsconnector/connector_test.go

```diff
@@ -0,0 +1,355 @@
+	connector, err := factory.CreateTracesToMetrics(context.Background(), settings, cfg, next)
+	require.NoError(b, err)
+	inputTraces, err := golden.ReadTraces("testdata/traces/traces.yaml")
+	require.NoError(b, err)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, connector.ConsumeTraces(context.Background(), inputTraces))
+	}
+}
+
+func BenchmarkConnectorWithMetrics(b *testing.B) {
+	factory := NewFactory()
+	settings := connectortest.NewNopSettings()
+	settings.TelemetrySettings.Logger = zaptest.NewLogger(b, zaptest.Level(zapcore.DebugLevel))
+	next, err := consumer.NewMetrics(func(context.Context, pmetric.Metrics) error {
+		return nil
+	})
+	require.NoError(b, err)
+
+	cfg := &config.Config{Datapoints: testMetricInfo(b)}
+	require.NoError(b, cfg.Unmarshal(confmap.New())) // set required fields to default
+	require.NoError(b, cfg.Validate())
+	connector, err := factory.CreateMetricsToMetrics(context.Background(), settings, cfg, next)
+	require.NoError(b, err)
+	inputMetrics, err := golden.ReadMetrics("testdata/metrics/metrics.yaml")
+	require.NoError(b, err)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, connector.ConsumeMetrics(context.Background(), inputMetrics))
+	}
+}
+
+func BenchmarkConnectorWithLogs(b *testing.B) {
+	factory := NewFactory()
+	settings := connectortest.NewNopSettings()
+	settings.TelemetrySettings.Logger = zaptest.NewLogger(b, zaptest.Level(zapcore.DebugLevel))
+	next, err := consumer.NewMetrics(func(context.Context, pmetric.Metrics) error {
+		return nil
+	})
+	require.NoError(b, err)
+
+	cfg := &config.Config{Logs: testMetricInfo(b)}
+	require.NoError(b, cfg.Unmarshal(confmap.New())) // set required fields to default
+	require.NoError(b, cfg.Validate())
+	connector, err := factory.CreateLogsToMetrics(context.Background(), settings, cfg, next)
+	require.NoError(b, err)
+	inputLogs, err := golden.ReadLogs("testdata/logs/logs.yaml")
+	require.NoError(b, err)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		require.NoError(b, connector.ConsumeLogs(context.Background(), inputLogs))
+	}
+}
+
+// testMetricInfo creates a metric info with all metric types that could be used
+// for all the supported signals. To do this, it uses common OTTL funcs and literals.
+func testMetricInfo(b *testing.B) []config.MetricInfo {
+	b.Helper()
+
+	return []config.MetricInfo{
+		{
+			Name:        "test.histogram",
+			Description: "Test histogram",
+			Unit:        "ms",
+			IncludeResourceAttributes: []config.Attribute{
+				{
+					Key: "resource.foo",
+				},
+				{
+					Key:          "404.attribute",
+					DefaultValue: "test_404_attribute",
+				},
+			},
+			Attributes: []config.Attribute{
+				{
+					Key: "http.response.status_code",
+				},
+			},
+			Histogram: &config.Histogram{
+				Buckets: []float64{2, 4, 6, 8, 10, 50, 100, 200, 400, 800, 1000, 1400, 2000, 5000, 10_000, 15_000},
+				Value:   "Double(1.4)",
```

Comment:
We can, but we can't use it directly as float (like 1.4) in the yaml file and we have to cast it to string -- because parser accepts string. I was debugging something and left all the `Double` and `Int` lying around 🤦 . Will fix it.